testcases/pkcs11/sess_perf.c

```diff
@@ -209,10 +212,12 @@ int do_SessionPerformance(unsigned int count)
         rc = finalize_des_encrypt_context(t[i].hsess);
         if (rc == FALSE) {
             show_error("finalize_aes_encrypt_context", (CK_RV) 0);
+            free(t);
             return FALSE;
         }
     }
 
+    free(t);
```

Comment:
Please use the following function exit:
```
    rv = TRUE;
ret:
    if (t != NULL)
        free(t);
    return rv;
```

Initialize 'rv = FALSE' and replace all
```
free(t);
return FALSE;
```
by
```
goto ret;
```

Reply:
OK, I am now reusing variable rc and use the ret goto lable.

Reply:
okay for me if it works but usually its better to have two local vars, one for caller return code (pre-set to eg false) and one to catch and check the callee return codes.